helix-core/src/main/java/org/apache/helix/task/WorkflowDispatcher.java

```diff
@@ -290,7 +297,6 @@ private void scheduleJobs(String workflow, WorkflowConfig workflowCfg,
         // Time is not ready. Set a trigger and update the start time.
         // Check if the job is ready to be executed.
         if (System.currentTimeMillis() >= workflowCtx.getJobStartTime(job)) {
-          scheduleSingleJob(job, jobConfig);
```

Comment:
Can remind me why scheduleSignleJob is removed? We already schedule the job in other places?

Reply:
The only line that was kept in scheduleSingleJob() was the line about creating job context, however, that is already done in processJob. You could check my comment in scheduleSingleJob.